src/main/java/com/google/sps/servlets/BooksServlet.java

```diff
@@ -0,0 +1,148 @@
+    Map<String, Object> books = new HashMap<>();
+    books.put("books", booksMap);
+    String json = gson.toJson(books); 
+    return json;
+  }
+
+ /**
+  * Creates list of Book objects from JSON results of Google Books query.
+  * @return list of book objects.
+  */
+  private List<Book> makeBooksList(String jsonString) {
+    JsonObject responseObject = new JsonParser().parse(jsonString).getAsJsonObject();
+    JsonArray booksJsonArray = responseObject.getAsJsonArray("items");
+    // Reduce number of books displayed if there less books in the json array than as default.
+    int numResults = booksJsonArray.size();
+    numResults = Math.min(numResults, NUM_BOOKS_PER_TERM);
+    List<Book> books = new ArrayList<>();
+    for (int i = 0; i < numResults; i++) {
+      JsonObject bookJson = booksJsonArray.get(i).getAsJsonObject().getAsJsonObject("volumeInfo"); 
+      String title = bookJson.get("title").getAsString();
+      String link = bookJson.get("infoLink").getAsString();
+      String image = bookJson.getAsJsonObject("imageLinks").get("thumbnail").getAsString();
+      String description = bookJson.get("description").getAsString();
+      String author = formatAuthors(bookJson.getAsJsonArray("authors"));
+      books.add(new Book.Builder(title, link).withImage(image).withDescription(description).withAuthor(author).build());
+    }
+    return books;
+  }
+
+ /**
+  * Reads in JSON from an open URL.
+  * @return JSON in String format.
+  */
+  private String readJsonFile(URL url) throws IOException {
+    Scanner jsonReader = new Scanner(url.openStream());
+    String data = "";
+    while(jsonReader.hasNext()) {
+        data += jsonReader.nextLine();
+    }
+    jsonReader.close();
+    return data;
+  }
+
+ /**
+  * Formats authors into string from JSON array.
+  * @return authors in String fomat.
+  */
+  private String formatAuthors(JsonArray authorsArray) {
+    String returnString = "";
+    int length = authorsArray.size();
+    for (int i = 0; i < length - 1; i++){
+      returnString += authorsArray.get(i).getAsString() + ", ";
+    }
+    returnString += authorsArray.get(length - 1).getAsString();
+    return returnString;
+  }
+
+ /**
+  * Gets string of JSON of Google Books API's response to a query including the term.
+  * @return String of Google Books API JSON response. 
+  */
+  private String getJsonStringForTerm(String term) {
+    String path = "https://www.googleapis.com/books/v1/volumes?";
+    String queryParam = "q=" + encodeTerm(term); 
+    String queryPath = path + queryParam;
+    try {
+      URL url = new URL(queryPath);
+      HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+      connection.setRequestMethod("GET");
+      connection.connect();
+      int responseCode = connection.getResponseCode();
+      if (responseCode == 200) {
+        return readJsonFile(url);
+      } else {
+        System.err.println("Error: connection response code is: " + responseCode);
```

Comment:
if (responseCode != 200) { 
  System.err.println("Error: connection response code is: " + responseCode);
}
return readJsonFile(url);